### DISCUSSION TITLE
Paket AVM-firewall ist und bleibt verschollen

### DISCUSSION THREAD

**Discussion by karstengit** [Freetz-NG/freetz-ng#657] [Allgemein]
Beschreibung siehe Bug https://github.com/Freetz-NG/freetz-ng/issues/654

Wenn es hier keine Antwort gibt ist Freetz-NG leider bezogen auf die Einflußnahme der AVM-firewall für die Tonne.

**Comment by karstengit** [comment]
**Es handelt sich hier definitiv um einen Bug.**

Durch gezieltes abschätzen kann man in die **.config** manuell folgenden Eintrag machen, **nachdem menuconfig beendet worden ist**:

```
#
# Web interfaces
#

#
# Addhole (needs dnsmasq)
#
FREETZ_PACKAGE_AUTHORIZED_KEYS=y
FREETZ_PACKAGE_AVM_FIREWALL=y
```

Mit dem Eintrag **FREETZ_PACKAGE_AVM_FIREWALL=y** wird die AVM-Firewall hinein kompiliert.

```
installing packages
  syslogd-cgi-0.2.3 .................    3.21 Kb 
  inetd-0.2 .........................    3.17 Kb 
  authorized-keys-0.1 ...............    2.64 Kb 
  avm-firewall-2.0.4_rc5 ............   16.06 Kb 
  haserl-0.9.36 .....................   10.32 Kb 

```

**Comment by aus-sua** [comment]
Ernsthaft!?

Leider hast Du weder das nun schon oft benutzte
RTFM
beachtet, noch Dir die anderen Hilfen/FAQ dazu
durchgelesen (z.B. FREETZ_AVM_VERSION_05_5X_MAX [=y] ).

Wenn Du mit rumstümpern (!) fertig bist und **wirklich** Interesse
an freetz/freetz-NG hast, dann melde Dich 'halt noch einmal;
ggf. hat ja jemand mit großer Frustrationstoleranz dann Zeit
und Lust, Dir da weiter zu helfen.

**Comment by karstengit** [comment]
Wenn eine passende Erklärung und Informationen zu finden wäre, ist ein rumstümpern nicht erforderlich.
Da Rückfragen nur mit dummer Anmache beantwortet werden, ist das leider ebenfalls keine Lösung.

Offensichtlich sind hier unterschiedliche Vorstellungen von Kommunikation gegeben.

Das FREETZ_AVM_VERSION_05_5X_MAX führt lediglich zu diesem Treffer 
https://www.ip-phone-forum.de/threads/nutzung-des-avm-firewall-forwarding-web-if-bei-fritz-os-7-x.300539/
der für eine andere Box gilt und bislang als Hinweis über eine Suchmaschine nicht auffindbar gewesen ist.

Warum kann bei dem Paket in der Beschreibung nicht einfach der Hinweis stehen, daß dieses nur bis zur Firmware XX.XX funktioniert?

Da AVM offensichtlich keine brauchbare integrierte Firewall mehr haben möchte, wäre es sehr interessant auf eine ältere Firmware-Version zurückzugehen, was jedoch nicht mehr möglich ist, da die passenden Dateien nicht mehr zur Verfügung stehen.

**Reply by aus-sua** [reply]
Was für ein rumjammern und nach Ausreden suchen!

> Warum kann bei dem Paket in der Beschreibung nicht einfach der Hinweis stehen, daß dieses
> nur bis zur Firmware XX.XX funktioniert?

Ganz einfach:
Weil es keiner in die Doku bzw. FAQ reinschreibt! 
An statt weiter rumzuheulen, mache es doch einfach selbst!
Mach ein pull oder editiere die Doku!
---> freetz-NG ist ein kostenloses **mit**-mach-Projekt!

Aber,
menuconfig nimmt Dir  (Dank der Entwickler!) die Mühe der AVM-Versions-Vergleiche ab!
Es bietet Dir eben nur die Pakete an, die auch mit Deiner VORHER festgelegten AVM-Version
wohl ohne Probleme funktionieren ---> RTFM
und (auch wichtig, zusätzlich!)
sowohl in der Hilfe der menuconfig als auch in der FAQ/Doku stehen z.B. solche Sachen:
zum avm-firewall:
 FREETZ_AVM_VERSION_05_5X_MAX [=y]
(da habe ich es auch her! per /)
zum iptables:
 Funktioniert iptables/nat/conntrack?
  IPtables funktioniert überall. Allerdings wird in Fritz!OS 05.2x und 05.5x von AVM der PacketAccelerator
 genutzt. Durch diesen funktioniert conntrack nicht mehr. Und damit nat/masquerading/state /transparenter
 Proxy und alles was sonst noch von conntrack abhängt.
 Abhilft schafft ein Downgrade auf eine alte Firmware, zB bei der 7270 ein “Alien” mit 04.88...

Damit steht da ja:
quasi ist iptables seit AVM-Version > 5.00 bei FRITZ!Boxen tot...!
---> RTFM! jo, immer wieder...
(ganz tot ist es wohl nicht, die Wege, um es doch ev. nutzen zu können, stehen auch in der FAQ)
Wenn Du es so testen möchtest, liebend gerne,
---> RTFM!

zu Debian:
es wurde Dir gesagt, mit Debian gibt es Probleme, die aber keiner so genau benennt...
Was machst Du:
Nichts! Du lieferst auch nur ein "es geht nicht!"
> ... startete der Kompiliervorgang, aber brach dann mit extrem unverständlichen
> Fehlermeldungen ab...

Hattest Du dazu die PREREQUISITES überhaupt gelesen und bedacht und beachtet?
Du lieferst keine Fehlermeldungen, keinen Log, nichts, nur lauter laute Luft!
fmake kennst Du? nein --> RTFM!
Ich hoffe, Du fragst da jetzt nicht noch einmal:
Aber warum steht zu Debian nichts in der Doku?
---> s.o. ab: Weil es keiner in die Doku bzw. FAQ reinschreibt!...

Aber laut Bug schreien und sich sowas von beleidigt fühlen, wenn man
dann eben ein RTFM um die Ohren bekommt!
Junge, junge,
O tempora, o mores!

PS:
Nur einmal so,
weil Du das ja auch nicht genannt hast (braucht ja auch keiner...), um welche Box
geht es Dir denn? 7590/7430/7390?
Weil:
> wäre es sehr interessant auf eine ältere Firmware-Version zurückzugehen

Welche der Boxen wäre es denn dann, die jemals eine Firmware kleiner v.5.00 hatten?

**Reply by fda77** [reply]
Danke, so ähnlich hätte ich das auch geschrieben ..

**Comment by karstengit** [comment]
Danke für die ausführliche Antwort und nun akzeptable Höflichkeit.

> Was für ein rumjammern und nach Ausreden suchen!

Dies ist eine Frage der Erwartungen und Perspektiven, welche unterschiedlich sind.

Ihr scheint als Entwickler vorauszusetzen, daß jeder der Eure Entwicklungsumgebung benutzt über ein tiefes Wissen verfügt und den Anspruch hat Entwickler zu sein. Dem ist (leider) nicht so.
Es sollte Wissen über Posix und das kompilieren von Software vorausgesetzt werden, jedoch ist der Build-Prozess dieser Firmware sehr komplex, vor allem was die Abhängigkeiten betrifft.
Insofern wird die Kritik in einigen Punkten angenommen.

Andererseits wird der Großteil der Benutzer Freetz aus reiner Benutzersicht benutzen wollen, ohne den Anspruch zu haben sowohl diese Entwicklungsumgebung, also auch viele Pakete genau zu verstehen. Dazu zähle ich mich dazu.
Wenn eine Auswahl fertiger Freetz-Firmware existieren würde, wäre ein eigenes Build wahrscheinlich nicht von Interesse.

Wenn Ihr Eure Kompetenz als Maßstab für alle anderen für eine Mitarbeit zu Grunde legt, wird es daher sehr schwierig.
(Dies ist sowohl ein Kompliment als auch ein Anstoß zum nachdenken.)

> Weil es keiner in die Doku bzw. FAQ reinschreibt!
> An statt weiter rumzuheulen, mache es doch einfach selbst!
> Mach ein pull oder editiere die Doku!
> ---> freetz-NG ist ein kostenloses mit-mach-Projekt!

Es ist mir nicht klar wie ein editieren mit github möglich ist. (github selber ist ein System das nicht sehr benutzerfreundlich ist.)
Ist dies ohne spezielle (Schreib-)Rechte überhaupt möglich?

> Damit steht da ja:
> quasi ist iptables seit AVM-Version > 5.00 bei FRITZ!Boxen tot...!
> ---> RTFM! jo, immer wieder...
> (ganz tot ist es wohl nicht, die Wege, um es doch ev. nutzen zu können, stehen auch in der FAQ)
> Wenn Du es so testen möchtest, liebend gerne,

So richtig klar ist es mit der Beschreibung leider immer noch nicht.
Die AVM-Firewall ist seit der Version 7 nicht mehr nutzbar? (Würde dies mit V 6.8 noch funktionieren?)
Für iptables gibt es noch eine Chance wenn conntrack rausgepatcht wird?

Ehrlich gesagt ist die unbefriedigende Firewall von AVM die Hauptmotivation für Freetz (neben Telnet mit root-Rechten).
Es ist bald nicht mehr nachvollziehbar warum überhaupt noch Sourcen von AVM zur Verfügung gestellt werden, wenn gleichzeitig immer mehr dafür getan wird, das eine Änderung der Firmware kaum noch möglich ist.

Für viele Benutzer von Freetz dürfte noch im Vordergrund stehen, daß die Fritzbox als Server-Ersatz benutzt wird.
Hier ist noch ein eigenständiger Server hinter der Fritzbox vorhanden und es stellt sich die Frage, ob sämtliche Dienste nicht besser auf diesem realisiert werden und die Fritzbox nur noch als Modem dient. Dies wäre natürllich Schade, denn derzeit ist praktisch eine doppelte Firewall realisiert.

> zu Debian:
> es wurde Dir gesagt, mit Debian gibt es Probleme, die aber keiner so genau benennt...
> Was machst Du:
> Nichts! Du lieferst auch nur ein "es geht nicht!"

Es wurden in dem Bug https://github.com/Freetz-NG/freetz-ng/issues/654 bereits die Fehlermeldungen ergänzt.
Da der Build-Prozess startet ist es möglich das nicht mehr allzuviele Dinge fehlen.

> Hattest Du dazu die PREREQUISITES überhaupt gelesen und bedacht und beachtet?

Es wurde bereits dokumentiert das die Paket-Empfehlung von Kubuntu verwendet worden ist.
Die aufgelisteten Pakete wurden nachfolgend auf Grund der Fehlermeldungen nachinstalliert.

> Aber laut Bug schreien und sich sowas von beleidigt fühlen, wenn man dann eben ein RTFM um die Ohren bekommt!

Beleidigend ist nur die Art und Weise mit ernst gemeinten Fehlermeldungen umzugehen.
(in Bug https://github.com/Freetz-NG/freetz-ng/issues/655 ist immer noch nicht klar wie das fehlende Kernelmodul sha1_generic.ko in das Build hineinkommen soll.)
Es wird um Verzeihung dafür gebeten, daß die meisten davon aus Unkenntnis der Zusammenhänge entstanden sind.
Aus diesen Gründen wurde auch die Kubuntu-VM aufgesetzt.

> Nur einmal so,
> weil Du das ja auch nicht genannt hast (braucht ja auch keiner...), um welche Box
> geht es Dir denn? 7590/7430/7390?

Dies wurde ebenfalls benannt und somit scheinen beide Seiten wichtige Details schon Mal zu überlesen. ;-)
Es geht um eine 7430 und eine 7360v2, wobei der Fokus auf der 7430 liegt.

**Comment by aus-sua** [comment]
ok, machen wir weiter...
Vorweg eins,
ich bin z.Z. auf Arbeit/zur Schicht z.Z. ist es damit zeitmäßig eher schlecht, ich bin auch 'eh nicht der
schnellste Schreiber/Denker, es kann da mit Antworten zeitmäßig immer dauern!

Vorweg zwei,
wir sollten uns erst einmal einigen, wie schon [gesagt](https://github.com/Freetz-NG/freetz-ng/discussions/624#discussioncomment-4306517):
> ... Also was ist Dein Problem und immer schön eines nach dem anderen und
> nichts vermischen und dann 100 Probleme auf einmal...

da war Deine Antwort:
https://github.com/Freetz-NG/freetz-ng/issues/654

Leider hast Du dann doch unter Deinen verschiedenen Posts einzeln weiter gemacht.
Ich habe einfach keine Lust und keine Zeit mir Deine Infos unter Deinen 10 verschiedenen Posts
zusammen zu suchen bzw. diese da auch noch weiter zu beobachten, sorry...

Also, Vorschlag:
Du eröffnest eine neue Discussion mit aussagefähigen Titel und darin z.B. den Links zu all' Deinen
vielen Posts (+ unter den "alten" Posts einen (abschließenden) gegenlink, quasi so etwas
wie "hier geht es dann weiter...")
oder
wir machen (nur) hier weiter (dann eben die Linkliste hier rein und auch Gegenlinks dort)

und dann eben auch nur
> ... immer schön eines nach dem anderen!

kurz zu Deinem o.g. Post:
> Es ist mir nicht klar wie ein editieren mit github möglich ist.

[pull request](https://github.com/Freetz-NG/freetz-ng/pulls) (wie? git, RTFM - ja, ist kein Running Gag!)
oder einfach erst einmal eine neue Discussion mit Deinen Änderungen/Beschreibungen der Hilfen/Dokus...

> Die AVM-Firewall ist seit der Version 7 nicht mehr nutzbar?

Nein, für **iptables** ist der Knackpunkt (lies selbst, ab Symbols:):
![hilfe](https://user-images.githubusercontent.com/73757472/205728039-a4790656-7bdd-4ff0-bea6-1c57944ceadc.JPG)

Ja, das läuft dann für **iptables** hinaus auf: Firmware < 7.0
Wie bin ich zu dem Bild gekommen?
Das ist die menuconfig-Hilfe zu iptables (jo, RTFM!)

Wie habe ich iptables gefunden?
 s.o. ab:
> ... menuconfig nimmt Dir (Dank der Entwickler!) die Mühe...

und dann die Firmwareversion ausgewählt und nachgesehen, bei welcher FW-Version eben
iptables noch bzw. nicht auswählbar ist.

Hat das jemand schon getestet, auch gegen alle AVM-Funktionen?
Lt. Deinem selbst gefundenen [Link ](https://www.ip-phone-forum.de/threads/nutzung-des-avm-firewall-forwarding-web-if-bei-fritz-os-7-x.300539/) wohl eher nicht, das wäre dann eben Deine Aufgabe...

Wo ist der Knackpunkt für die AVM-firewall ?
s.o. bei:
FREETZ_AVM_VERSION_05_5X_MAX [=y] 

Wie finde ich das "FREETZ_AVM_VERSION_05_5X_MAX [=y]".
RTFM (Tipp: in der Hilfe der menuconfig für die AVM-firewall)

Aber ich habe doch keine AVM-firewall unter "Packages > Web interfaces"?
RTFM/doku (Tipp: 7390 - wähle halt eine Box + Firmware von vor mind. 10 Jahren aus
(so lange steht es ja schon mind. in der FAQ/Hilfe!))!
Teste doch auch einmal die Suche per: /

> noch nicht klar wie das fehlende Kernelmodul sha1_generic.ko in das Build hineinkommen soll

RTFM, nun ja, Bildschirm-error-Hilfe gelesen?
![error(M)](https://user-images.githubusercontent.com/73757472/205737843-5af48670-9c4f-482d-b55f-dc4635ae2a31.JPG)

--> Tipp: buchstäblich! + wortwörtlich!
also:
Enable missing with 'make kernel-menuconfig' as (M)odule (and push a pull request). Use '/' to search in menuconfig
also:
![hilfe(M)](https://user-images.githubusercontent.com/73757472/205741706-67047938-6e28-4b05-8740-cc6fa689bbc9.JPG)

---> make kernel-menuconfig
---> Enable missing with as (M)odule (Tipp: nicht (x) ! )  ---> sorry, RTFM, bzw. auch: Bildschirm ablesen
---> Use '/' to search missing in menuconfig
---> push a pull request

Dabei gibt es dann wirklich ggf. Probleme/Ungereimtheiten (lösbar), aber wenn Du dann darauf stößt,
denn müssen wir da eben @fda77 genauer anfragen!

> Dies wurde ebenfalls benannt und somit scheinen beide Seiten wichtige Details schon Mal zu
< überlesen. ;-) Es geht um eine 7430 und eine 7360v2, wobei der Fokus auf der 7430 liegt

OK, 7430 + ggf. 7360v2
Wobei:
wurde die 7360v2 zeitlich nicht erst nach meinem Post bekannt gegeben?
gab es wirklich verbindliche Ansagen: ich habe eine 7340 und möchte die mit ... bestücken?
Zumindest für mich klang das alles wie: Testbeispiel-Box?!
meine Ausrede: aber, s.o.  bei: --> unter Deinen 10 verschiedenen Posts...
Also, ok: 7430 + 7360v2 :-)

PS:
Ich bin kein Entwickler, ich bin genau so Anwender wie Du, ggf. sind meine Linux-Kenntnisse
sogar geringer als Deine, aber:
Ich bin alt, ich lese wirklich "Handbücher"!...

PS2:
Vorweg drei:
Mir geht es dann/da auch um die Debian-PREREQUISITES, es wäre schön, wenn wir das dann auch noch mit erledigen...

PS3:
Frage, weil es auch noch nicht direkt "geliefert" wurde:
Wo für brauchst Du iptables/AVM-firewall?
Für ein reines forwarding gibt es ja z.B.:
[avm-portfw](https://freetz-ng.github.io/freetz-ng/make/avm-portfw.html)

PS4:
Ältere Firmware-Versionen kann man z.B. durchaus auch freundlich [hier](https://www.ip-phone-forum.de/threads/suche-firmware-recover-images.312873/) erfragen (RTFM!, das dortige...)

**Comment by karstengit** [comment]
> ich bin z.Z. auf Arbeit/zur Schicht z.Z. ist es damit zeitmäßig eher schlecht

Die Angelegenheit eilt nicht.

> Leider hast Du dann doch unter Deinen verschiedenen Posts einzeln weiter gemacht.

Macht es nicht Sinn zu versuchen die zusammengehörigen Dinge zusammen zu halten?
Das Thema Debian 11 ist nun unter https://github.com/Freetz-NG/freetz-ng/issues/654 und der andere Bug wurde für weitere Postings gesperrt.



>>    Die AVM-Firewall ist seit der Version 7 nicht mehr nutzbar?

> Nein, für iptables ist der Knackpunkt (lies selbst, ab Symbols:):
> Ja, das läuft dann für iptables hinaus auf: Firmware < 7.0

Wegen FREETZ_KERNEL_VERSION2 = n ?
Die Abhängigkeiten sind leider meistens nur dann klar zu erkennen, wenn sie bereits verstanden worden sind.

Wie auch immer, es läuft also darauf hinaus eine ältere Version zu bauen.
Damit kommt das Problem auf, daß die Dateien dafür von AVM nicht (mehr) bereitgestellt werden !?

Für die ältere Firmware wurde ein Anfrage an AVM gestellt und folgendermaßen beantwortet:

> Ich erfrage gerne intern, ob wir die Version 6.83 für die FRITZ!Box 7430 bereitstellen. 
> Sobald mir neue Informationen vorliegen, melde ich mich umgehend bei Ihnen. Bitte haben Sie bis dahin ein wenig Geduld.

Andere Quellen für eine ältere Firmware sind leider noch nicht aufgetaucht und wahrscheinlich fehlt dann noch ein passendes Quellpaket für die Box aus dieser Zeit?

> Hat das jemand schon getestet, auch gegen alle AVM-Funktionen?
> Lt. Deinem selbst gefundenen [Link ](https://www.ip-phone-forum.de/threads/nutzung-des-avm-firewall-forwarding-web-if-bei-fritz-os-7-x.300539/)wohl eher nicht, das wäre dann eben Deine Aufgabe...

Das wäre nicht das Problem, sobald ein passendes Image gebaut und geflasht ist.

> OK, 7430 + ggf. 7360v2
> Wobei:
> wurde die 7360v2 zeitlich nicht erst nach meinem Post bekannt gegeben?

Es wurde lediglich die Frage exakt beantwortet.
O.K. Es ist so das derzeit eine 7360v2 mit einer Original-Firmware 124.06.52 läuft um DSL bereitzustellen.

Damit das Internetz verfügbar bleibt wurde die 7430 zum experimentieren angeschafft.
Auf dieser ist zur Zeit eine Original-Firmware 146.06.83.
AVM stellt nur noch die Firmware FRITZ.Box_7430-07.29.image für diese Box zur Verfügung, welche eine (unbrauchbare) Firewall hat.
So lange keine Firmware-Datei 06.83 zur Verfügung steht, macht es keinen Sinn ein Freetz zu flaschen, weil dann nur noch eine Firmware V7 möglich ist ohne Chance auf Rückkehr zu der (brauchbaren) älteren Firmware Version.

> RTFM, nun ja, Bildschirm-error-Hilfe gelesen?
> --> Tipp: buchstäblich! + wortwörtlich!
> also:
> Enable missing with 'make kernel-menuconfig' as (M)odule (and push a pull request). Use '/' to search in menuconfig

Die Suche nach  sha1_generic.ko ist erfolglos, sowohl im menuconfig von freetz als auch in der Kernel Configuration:

![Screenshot_20221206_141245](https://user-images.githubusercontent.com/5941550/205922560-91d35304-0540-4fb5-8173-2b960dce17af.png)

> Dabei gibt es dann wirklich ggf. Probleme/Ungereimtheiten (lösbar), aber wenn Du dann darauf stößt,
denn müssen wir da eben @fda77 genauer anfragen!

@fda77 - Ein Tip für einen alten blinden Mann?

> Vorweg drei:
> Mir geht es dann/da auch um die Debian-PREREQUISITES, es wäre schön, wenn wir das dann auch noch mit erledigen...

Diese sind wahrscheinlich ein lösbares Problem, wenn ich eine Idee habe wie der Fehler

```
In file included from pseudo_wrappers.c:297:
pseudo_wrapfuncs.c:13898:1: error: conflicting types for ‘statx’
13898 | statx(int dirfd, const char *path, int flags, unsigned int mask, struct statx *statxbuf) {
      | ^~~~~

```
zu umschiffen ist?

> Frage, weil es auch noch nicht direkt "geliefert" wurde:
> Wo für brauchst Du iptables/AVM-firewall?

https://freetz-ng.github.io/freetz-ng/make/avm-firewall.html
Mit AVM-FIREWALL-CGI kann man direkt über eine Weboberfläche Folgendes tun:

- Die von AVM vordefinierten Firewallregeln ändern/löschen
- Eigene Regeln hinzufügen (sehr praktisch um lästiges “nach Hause telefonieren” von mancher Software im Router zu unterbinden)
- Portforwardings erstellen, die auch auf die Fritz selber (0.0.0.0) zeigen können.(z.B. für FTP oder webserver Freigaben ins I-Net)

Der Teil für das Internet ist dabei nicht interessant.

https://freetz-ng.github.io/freetz-ng/make/iptables-cgi.html

Iptables läuft mit noch höherer Priorität als diese AVM-Firewall und gibt damit volle Kontrolle über den Netzwerkverkehr.

**Reply by fda77** [reply]
Nimm einfach fakeroot statt pseudo

Wie gesagt, keine Ahnung was mit Debian ist, nur dass es da ständig Problem mit gibt

> Ein Tip für einen alten blinden Mann?

Genau deshlab sollen Anfänger die "Doku" schreiben! Die Kernelmodule sind natürlich ohne .ko im menuconfig des kernels ^^

iptables geht nur eingeshränkt, fürs telefonieren gibts addhole filterlisten

**Reply by karstengit** [reply]
Danke für die Antwort und eine kurze Zwischenmeldung wegen Zeitmangel.

> Nimm einfach fakeroot statt pseudo

Was bedeutet das bitte ?
fakeroot ist bekannt, aber in welchem Kontext soll es benutzt werden?
Was ist mit pseudo gemeint?

> Genau deshlab sollen Anfänger die "Doku" schreiben! Die Kernelmodule sind natürlich ohne .ko im menuconfig des kernels ^^

O.K. Dann muß dies noch einmal wiederholt werden.
Ist irgendwo eine Beschreibung zu finden wie die Doku aktualisiert werden kann?

Bezüglich Debian 11 denke ich das die Quelltexte der Header-Dateien unterschiedlich zu Kubuntu sind.
Das muß ich noch Mal vergleichen, wenn ich diese gefunden habe.

> iptables geht nur eingeshränkt, fürs telefonieren gibts addhole filterlisten

Das wäre zumindest schon Mal ein hilfreicher Anfang.

Inzwischen hat AVM geantwortet:

> vielen Dank für Ihre Geduld.
>  
> In der Regel stellen wir ältere FRITZ!OS-Versionen nicht zum Download bereit. Aus Sicherheitsgründen sollte immer die aktuelle FRITZ!OS-Version verwendet werden. Daher können wir Ihnen die Version 6.83 nicht zur Verfügung stellen.
>  
> Die älteste verfügbare Version ist FRITZ!OS 7.12, die in einigen Fällen als Zwischenupdate notwendig ist:
>  
>     [FRITZ!Box meldet "Die angegebene Datei enthält eine Firmware-Version, die älter ist als die momentan aktive Firmware"](https://avm.de/service/wissensdatenbank/dok/FRITZ-Box-7430/3657_FRITZ-Box-meldet-Die-angegebene-Datei-enthalt-eine-Firmware-Version-die-alter-ist-als-die-momentan-aktive-Firmware/)
>  
> Ich wünsche Ihnen einen schönen Tag.

Somit bleibt die 7430 weiterhin erst einmal auf der älteren Original-Firmware, da sich diese nicht mehr wiederherstellen läßt.
Da hier keiner zu dem Beschaffungsproblem der Firmware geantwortet hat, muß ich woanders schauen eine ältere Version aufzutreiben. Die Firmware Version 7 ist aus meiner Sicht leider nicht mehr zu gebrauchen.

**Comment by aus-sua** [comment]
Ach komm jetzt, ernsthaft?!
> Da hier keiner zu dem Beschaffungsproblem der Firmware geantwortet hat...

wie schon immer: --> RTFM!
Damit meine ich, lies doch einfach, was Dir geantwortet wird!
also, s.o.:  bei PS4:  der Link ist zum anklicken (kennste? ... Mit der Maus!)  ;-(

Und selbst wenn nicht,
Suchmaschinen findet mindesten eine Link, direkt zur Firmware!

Aber ich habe da dann doch schon wieder einmal Deine Arbeit gemacht und mich
zusätzlich auch noch anderweitig gekümmert (danke an's DEB):
--> https://web.archive.org/web/20171122054034/http://ftp.avm.de/fritz.box/fritzbox.7430/firmware/deutsch/
--> https://web.archive.org/web/20171122054035/http://ftp.avm.de/fritz.box/fritzbox.7430/x_misc/deutsch/
Ich würde Dir damit wirklich empfehlen,
zumind. einige Euros an die web.archive.org zu spenden!
Geht ganz schnell, z.B. Paypal!
Und übernimm denen dabei bitte die fees. Das gehört zum guten Ton!
Ernsthaft! :-o

Deinen Fetisch zu einer besonderen "geilen" Firmware verstehe ich sowie so nicht:
Was unterscheidet für Dich eine FW v.05.5x von der letzten FW 05.87 der 7360v2?
Warum muß! Du die v.05.5x behalten?

Was bedeutet das (zur 7340):
> Die Firmware Version 7 ist aus meiner Sicht leider nicht mehr zu gebrauchen.

???
Kann die 7340 nicht auch "dualboot/bootmanger"!? Kann man googlen oder eben auch
im menuconfig nach sehen... wäre dann ja aucvh wieder ---> RTFM

Alles andere später (am Abend, muß arbeiten...), aber es wird auch da nicht "besser" ;-)
z.B.:
>  fakeroot vs. pseudo

---> RTFM, oder z.B. per / in menuconfig...
oder, ganz einfach, zur "Erhellung",
wird immer wieder gerne genommen... die [News von freetz-NG](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/NEWS.md#2021-02-27)! 
Suchmaschinen wurden ja auch schon erfunden, google?
usw.... usw... 

PS:
Als Infobasis:
beschäftige Dich schon einmal mit:
- flashen der Box
- signierte Firmware
- Fehlermeldungen AVM
- Fehlermeldungen freetz/freetz-NG
- Flash-Partitionen sichern (werden wir wohl nicht mehr brauchen..., aber bitte lesen!)
- tools/push_firmware -h
usw... usw...
z.B. per ---> https://freetz-ng.github.io/freetz-ng/wiki/
oder Suchmaschinen...
was ja eben auch nur wieder ein:
--> RTFM!
ist...

PS2:
zur Info zu Deinem "Firmware-Fetisch":
--> https://www.ip-phone-forum.de/threads/7490-dualboot-was-passiert-mit-einstellungen.298062/
ja, is' 'ne andere Box, es geht aber um:
dort, ab dem Post 5 (zu meiner Frage: warum unbedingt 5.5x und nicht 5.83 der 7360v2)
AVM-update-Politik, Nutzung älterer Firmware usw...,
auch die Beschreibung im o.g. Link in PS4: lesen und verstehen (FW < 06.03...)

**Reply by karstengit** [reply]
> Ach komm jetzt, ernsthaft?!
> Damit meine ich, lies doch einfach, was Dir geantwortet wird!
> also, s.o.: bei PS4: der Link ist zum anklicken (kennste? ... Mit der Maus!) ;-(

Tja - vielleicht überliest man die wertvollen Hinweise einfach nur zwischen den ganzen Beleidigungen?

An sich ist das schon interessant, da ich bereits bei archive.org gesucht hatte und die Verzeichnisse gefunden hatte, in denen allerdings die Dateien nicht archiviert worden sind.

Vielleicht werde ich einfach zu alt für den Sch....

Na schön, dann kann es ja nun immerhin weiter gehen.

> Deinen Fetisch zu einer besonderen "geilen" Firmware verstehe ich sowie so nicht

Schade.

> was ja eben auch nur wieder ein:
> --> RTFM!
> ist...

Das ist in diesen Fällen tatsächlich berechtigt.
Wird wohl ein paar Tage dauern ...

**Comment by fda77** [comment]
Spam: https://github.com/Freetz-NG/freetz-ng/issues/654
